src/server/images/controllers.js

```diff
@@ -7,8 +7,26 @@ import {CreatedPostManager} from '../users/models';
 import assert from 'assert';
 
 export default {
-  get(req, res, cb) {
-    const {item, image} = req.query;
+  getById(req, res, cb) {
+    const imageKey = req.params.id;
+    // get an image
+    getPromise(imageKey)
+    .then((value)=>{
+      value.url = new S3Connector().getImageUrl(imageKey);
+      res.status(200).send(value);
+      cb();
+      return;
+    })
+    .catch((err)=>{
```

Comment:
It may cause some side-effect
like err.notFound. In this case we need to response with 400 statusCode. (Bad Request)
and the others is literally Internal Database Error